packages/aws-cdk-lib/aws-lambda-event-sources/lib/kafka.ts

```diff
@@ -55,6 +55,13 @@ export interface KafkaEventSourceProps extends BaseStreamEventSourceProps {
    * @default - discarded records are ignored
    */
   readonly onFailure?: lambda.IEventSourceDlq;
+
+  /**
+   * The time from which to start reading. Unix time seconds or `Date`
+   *
+   * @default - no timestamp
+   */
+  readonly startingPositionTimestamp?: number | Date;
```

Comment:
@moelasmar, here is one way to achieve it but is it very idiomatic to do it that way in CDK?

Reply:
in CDK we do not accept the union types in L2 any more. Personally I was thinking on replacing the type to be TimeStamp instead of number, but to be align with kinesis, I think we can add another property, and let customers choose which one to use, and for sure they can only use one of them.

Reply:
After digging into how this kind of property type change should happen in CDK, I found out that there is no clearly defined convention for it. There is a long discussion, e.g., on [Enhanced L1s RFC](https://github.com/aws/aws-cdk-rfcs/pull/657) about using the `_v2` pattern, but no consensus on going with it.

So, I suggest that we merge this like this:

> I can accept to have this property to keep it aligned with Kinesis

and worry about the API later once there is a clear convention on how to proceed.

The changes I made to test the union type are now reverted.